webpack/ForemanInventoryUpload/Components/TabHeader/TabHeader.js

```diff
@@ -1,26 +1,57 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Grid, Button, Icon } from 'patternfly-react';
+import { Grid, Button, Icon, DropdownItem, Dropdown, DropdownToggle, DropdownToggleAction } from 'patternfly-react';
 import { noop } from 'foremanReact/common/helpers';
 import { sprintf, translate as __ } from 'foremanReact/common/I18n';
 import { isExitCodeLoading } from '../../ForemanInventoryHelpers';
 import './tabHeader.scss';
 
-const TabHeader = ({ exitCode, onRestart, onDownload, toggleFullScreen }) => (
+const onActionToggle = () => {
+  setIsActionOpen(prev => !prev);
+};
+
+const dropdownItems = [
+  <DropdownItem
+    aria-label="restart_disconnected"
+    ouiaId="restart_disconnected"
+    key="restart_disconnected"
+    component="button"
+    onClick={restartDisconnected}
+    isDisabled={isExitCodeLoading(exitCode)}
+  >
+    {__('Restart without uploading')}
+  </DropdownItem>,
+];
+
+const TabHeader = ({ exitCode, onRestart, onDownload, restartDisconnected, toggleFullScreen }) => (
   <Grid.Row className="tab-header">
     <Grid.Col sm={6}>
       <p>{sprintf(__('Exit Code: %s'), exitCode)}</p>
     </Grid.Col>
     <Grid.Col sm={6}>
       <div className="tab-action-buttons">
         {onRestart ? (
-          <Button
-            bsStyle="primary"
-            onClick={onRestart}
-            disabled={isExitCodeLoading(exitCode)}
-          >
-            {__('Restart')}
-          </Button>
+          <Dropdown
+          aria-label="restart_dropdown"
+          ouiaId="restart_dropdown"
+          toggle={
+            <DropdownToggle
+              aria-label="restart_report_toggle"
+              ouiaId="restart_report_toggle"
+              splitButtonItems={[
+                <DropdownToggleAction key="action" aria-label="bulk_actions" onClick={onRestart}>
+                  {__('Restart')}
+                </DropdownToggleAction>,
+              ]}
+              splitButtonVariant="action"
+              toggleVariant="primary"
+              onToggle={onActionToggle}
+              isDisabled={isExitCodeLoading(exitCode)}
+            />
+        }
+          isOpen={isActionOpen}
+          dropdownItems={dropdownItems}
+        />
```

Comment:
You have already done a lot of work to fit this in, but if interested this could have been implemented in a separate component with "newer" technologies even without using redux actions reducers and all that stuff.

for the API call you could use `useAPI`

`import { useAPI } from 'foremanReact/common/hooks/API/APIHooks';`